engine/runtime/src/main/java/org/enso/interpreter/runtime/data/EnsoMultiValue.java

```diff
@@ -82,12 +375,17 @@ public final Object castTo(Type type) {
    */
   public final Pair<Function, Type> resolveSymbol(
       MethodResolverNode node, UnresolvedSymbol symbol) {
+    var ctx = EnsoContext.get(node);
+    Pair<Function, Type> foundAnyMethod = null;
     for (Type t : types) {
-      var fn = node.execute(t, symbol);
-      if (fn != null) {
-        return Pair.create(fn, t);
+      var fnAndType = node.execute(t, symbol);
+      if (fnAndType != null) {
+        if (fnAndType.getRight() != ctx.getBuiltins().any()) {
```

Comment:
So this is the only reason why we introduce the `Pair`?
I vaguely remember having a need for such info as well when dealing with `hasWarning` for `Any`. Maybe `resolveWarningFunction` could be simplified now?

Reply:
I did a bit of cleanup in 8954f5c921